src/main/scala/org/ergoplatform/nodeView/mempool/ErgoMemPool.scala

```diff
@@ -154,18 +170,16 @@ class ErgoMemPool private[mempool](pool: OrderedTxPool, stats : MemPoolStatistic
 
   /**
     * Get average fee for the specified wait time interval
-    * @param expectedWaitTimeMinutes maximal amount of time for which transaction can be kept in mempool
-    * @param txSize size of transaction (in bytes)
+    * @param expectedWaitTimeMinutes - maximal amount of time for which transaction can be kept in mempool
+    * @param txSize                  - size of transaction (in bytes)
     *  @return recommended fee value for transaction to be proceeded in specified time
     */
   def getRecommendedFee(expectedWaitTimeMinutes : Int, txSize : Int) : Long = {
     if (expectedWaitTimeMinutes < MemPoolStatisticsParams.nHistogramBins) {
       // locate first non-empty histogram bin preceding or equal to the specified wait time
       for (i <- expectedWaitTimeMinutes to 0 by -1) {
         val h = stats.histogram(expectedWaitTimeMinutes)
-        if (h.nTxns != 0) {
-          return h.totalFee / h.nTxns * txSize / 1024
-        }
+        if (h.nTxns != 0) h.totalFee / h.nTxns * txSize / 1024
       }
```

Comment:
This statement has no effect, so you simply miss computations inside it. It's better to avoid iterating over collection using scala `for`. Tre re-implement this piece of code using `map`  and `collectFirst`.